File: src/dialogs/aboutdialog.cpp
#include <QDesktopServices>
#include <QFile>

#include "dialogs/aboutdialog.h"
#include "ui_aboutdialog.h"

AboutDialog::AboutDialog(QWidget *parent): QDialog(parent), ui(new Ui::AboutDialog) {
	ui->setupUi(this);
	ui->label->setText(ui->label->text().replace("{VERSION}", VERSION));
}

AboutDialog::~AboutDialog() {
	delete ui;
}

void AboutDialog::on_closeAboutDialog_clicked() {
	close();
}

File: dialogs/aboutdialog.py
from PySide6.QtWidgets import QDialog
from PySide6.QtCore import QResource

from ui.ui_aboutdialog import Ui_AboutDialog

class AboutDialog(QDialog):
	ui: Ui_AboutDialog
	version: str
	licenseText: str

	def __init__(self, version:str, parent = None):
		super().__init__(parent)
		self.ui = Ui_AboutDialog()
		self.ui.setupUi(self)
		self.version = version
		self.ui.headerLabel.setText(self.ui.headerLabel.text().replace("{version}", self.version))
		res = QResource(":/LICENSE.txt")
		self.licenseText = res.uncompressedData().toStdString()
		self.ui.licenseLabel.setText(self.licenseText)
